content/en/docs/concepts/cluster-administration/api-priority-and-fairness.md

```diff
@@ -0,0 +1,138 @@
+mapped to it are described as _exempt_, because of the lack of
+concurrency limitation.
+
+For a non-exempt request, if it is not allowed to be handled
+immediately then the API server will either reject the request or hold
+it in a queue for a while.  Each non-exempt priority level has a
+number of FIFO queues.  Whenever that priority level's number of
+requests currently being handled drops below the concurrency limit,
+the API server draws requests to begin handling from these queues in a
+fair way.
+
+## Flow schema
+
+To determine how to control a request, an API server first classifies
+that request to a _flow_ according to a configured collection of _flow
+schemas_.  The flow schemas are considered in order, according to each
+flow schema's _matching precedence_ (which is a number).  A flow
+schema has rules that describe which requests match that schema.
+Matching takes into account (a) the requested operation&mdash;as
+identified by the HTTP request verb and URI&mdash;and (b) if the request
+arrived through a secured port then the authenticated user attributes
+otherwise just the UserAgent.  Among the schemas that match the
+request, one with the logically highest matching precedence wins.
+
+A flow schema also has a declaration of how to complete the _flow
+identifier_ for the request's flow.  A flow identifier is a pair of
+strings: the name of its flow schema plus one more string called the
+_flow distinguisher_.  A flow schema indicates whether the
+distinguisher is (1) the username of the client making the request,
+(2) the namespace (if any) of the object that the request seeks to
+access, or (3) the empty string.
+
+A flow schema also states the name of the priority level to use for
+requests that match.
+
+## Queuing
+
+An API server first classifies a request to a flow, and thus a
+corresponding priority level, and if the priority level is not exempt
+then the API server assigns the request to a queue using a technique
+called "shuffle sharding".  This technique starts with hashing the
+flow identifier, and then uses the hash value as a source of entropy
+to pseudo-randomly pick a small subset of the queues, and finally
+assigns the request to one of the shortest queues in that subset.  The
+priority level declares the size of the subset.
+
+After assigning a request to a priority level and a queue, the
+API server immediately begins handling the request if that would not
+violate the priority level's concurrency limit; otherwise the queue
+holds the request until it can be executed or is evicted.
+
+The API server applies two criteria to limit queue occupancy.  First,
+each non-exempt priority level has a queue length limit.  The
+API server will not add a request to a queue that is already at or
+exceeding its length limit; rather, the request is immediately
+rejected.  Second, each API server is configured with a limit on how
+long a request can wait in a queue.  If that limit expires before the
```

Comment:
[worth to mention a request is rejected only when the priority level's concurrency limit is exceeded](https://github.com/kubernetes/website/pull/15899/files#r318790307) (@yliaog)